pkgs/os-specific/darwin/apple-source-releases/Libsystem/default.nix

```diff
@@ -89,14 +99,24 @@ appleDerivation rec {
     done
 
     ld -macosx_version_min 10.7 -arch x86_64 -dylib \
-       -o $out/lib/libSystem.B.dylib \
+       -o $out/lib/libSystem.B.x86_64.dylib \
        -compatibility_version 1.0 \
        -current_version 1226.10.1 \
        -reexport_library $out/lib/system/libsystem_c.dylib \
        -reexport_library $out/lib/system/libsystem_kernel.dylib \
        -reexport_library $out/lib/libSystem_internal.dylib \
        $args
 
+    ld -macosx_version_min 10.7 -arch i386 -dylib \
+       -o $out/lib/libSystem.B.i386.dylib \
+       -compatibility_version 1.0 \
+       -current_version 1226.10.1 \
+       -reexport_library $out/lib/system/libsystem_c.i386.dylib \
+       -reexport_library $out/lib/system/libsystem_kernel.i386.dylib \
+       $args
+
+    lipo -create $out/lib/libSystem.B.x86_64.dylib $out/lib/libSystem.B.i386.dylib -output $out/lib/libSystem.B.dylib
```

Comment:
This doesn't play very well with ```stdenv/darwin/make-bootstrap-tools.nix```. In its ```build``` phase it will not pack ```libSystem.B.dylib```, instead remakes during the ```unpack``` phase. The problem is that other bootstrap tools binaries will depend on ```libSystem.B.x86_64.dylib``` which doesn't exist on the real darwin system. I guess I need to figure out a way to get rid of ```$out/lib/libSystem.B.x86_64.dylib``` and ```$out/lib/libSystem.B.i386.dylib``` which are needed when the gluing is done with ```lipo```.